### PR TITLE
[RFC] workflow: updated security-vulnerabilities-report

### PR DESCRIPTION
**Pull request by CamberLoid**
Topic Description
-----------------

This is the security vulnerabilities bug report, next version.
Contents are subjected to be discussed and change while being in draft status. Comments inside configuration are subjected to be removed when being ready-to-review.
While in draft status, the content is subject to be reviewed, changed and force-pushed frequently. 

Package(s) Affected
-------------------

N/A

Security Update?
----------------

Kind of

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 1]
```diff
@@ -1,34 +1,94 @@
+# This is the draft of security vulnerabilities bug report, next version.
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Leave these two lines for your PR comment.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 2]
```diff
@@ -1,34 +1,94 @@
...
+# Contents are subjected to be discussed and change.
 name: 'Security Vulnerabilities Report'
 description: 'Any security vulnerabilities report should be filed with this template.'
 labels: ['security']
 body:
+# This field is designed in case of the reporting issue is in a component of our package.
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: This field is designed for listing affected package(s) and component(s).
E.g.: libX11 in `x11-lib`.

No need for further explanation.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 3]
```diff
@@ -1,34 +1,94 @@
...
+# Severity is a new field of the template. This may help maintainers to identify which issue is most urgent and require attentions.
+- type: dropdown
+  attributes:
+    label: 'Severity'
+    description: | 
+      Select the severity of reporting vulnerabilities. Refer to our wiki for details.
+      Choose the highest in case of multiple vulnerabilities.
+    #Through there is no such wiki yet.
+    options:
+      - 'Critical'
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml:33]: Add a "0-day" option, meaning that the vulnerability is already being used for malicious activities.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 4]
```diff
@@ -1,34 +1,94 @@
...
+    placeholder: '(common name) `package_name` version'
   validations:
     required: true
-- type: textarea
+- type: input
   attributes:
-    label: 'Other security advisory IDs'
-    description: 'Other security advisory IDs. Please list with references to each ID.'
+    label: 'CVE IDs'
+    description: 'Comma-separated list of CVE IDs. Fill N/A if no CVE ID is assigned.'
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Fill in "N/A" if no CVE ID is assigned.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 5]
```diff
@@ -1,34 +1,94 @@
...
+    placeholder: 'CVE-20xx-xxxx[, CVE-20xx-xxxx, ...]'
   validations:
     required: true
+# Severity is a new field of the template. This may help maintainers to identify which issue is most urgent and require attentions.
+- type: dropdown
+  attributes:
+    label: 'Severity'
+    description: | 
+      Select the severity of reporting vulnerabilities. Refer to our wiki for details.
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: reported vulnerability(ies)

Comment (MingcongBai) [reply .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: What Wiki?

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 6]
```diff
@@ -1,34 +1,94 @@
...
+      Choose the highest in case of multiple vulnerabilities.
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Choose the highest severity out of the IDs listed.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 7]
```diff
@@ -1,34 +1,94 @@
...
+    options:
+      - 'Critical'
+      - 'High'
+      - 'Moderate'
+      - 'Low'
 - type: textarea
   attributes:
-    label: 'Description'
-    description: 'Original mailing list message(s) or other description here.'
+    label: 'Other security advisories'
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Other security advisory IDs

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 8]
```diff
@@ -1,34 +1,94 @@
...
+    description: 'Other security advisories, mailing list archive or original bug report. Please list with references to each ID.'
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Other security advisory(ies), mailing list message(s), or bug report(s).

Comment (MingcongBai) [reply .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: reference(s)

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 9]
```diff
@@ -1,34 +1,94 @@
...
+    placeholder: |
+      Examples:
+      - Debian: [DSA-xxxx-x](link_to_DSA);
+      - Links to original mail list archive;
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Mailing list message

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 10]
```diff
@@ -1,34 +1,94 @@
...
+      - Links to original issue, if applicable
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Bug report *

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 11]
```diff
@@ -1,34 +1,94 @@
...
   validations:
     required: true
 - type: textarea
   attributes:
-    label: 'Patches'
-    description: 'Enter "N/A" if version update is preferred, otherwise please label each patch with ref. to original vulnerability ID(s).'
+    label: 'Description'
+    description: |
+      Original mailing list message(s) or other description may list here.
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: may be listed here *

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 12]
```diff
@@ -1,34 +1,94 @@
...
   validations:
     required: true
 - type: textarea
   attributes:
-    label: 'Patches'
-    description: 'Enter "N/A" if version update is preferred, otherwise please label each patch with ref. to original vulnerability ID(s).'
+    label: 'Description'
+    description: |
+      Original mailing list message(s) or other description may list here. 
+      You may also include these following supplemental details:
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: You may also include supplemental details *

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 13]
```diff
@@ -1,34 +1,94 @@
...
+      - Bug introduced in which version
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Version where the bug was introduced

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 14]
```diff
@@ -1,34 +1,94 @@
...
+      - Possible security impact on AOSC OS
+      - Available PoCs (Required if the reporting target is AOSC-developed component)
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: "required," no need to capitalise first letter in parenthetical expressions

Comment (MingcongBai) [reply .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: is an AOSC-developed component

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 15]
```diff
@@ -1,34 +1,94 @@
...
+      - Available patches or fixed version for this specific CVE(s)
+      - How to fix that problem
+      - Linked CWE if available.
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: CWE (Common Weakness Enumeration)

Comment (MingcongBai) [reply .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Leave period point out - unless you want to use it on each list item.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 16]
```diff
@@ -1,34 +1,94 @@
...
+      You may also include these following supplemental details:
+      - Bug introduced in which version
+      - Possible security impact on AOSC OS
+      - Available PoCs (Required if the reporting target is AOSC-developed component)
+      - Available patches or fixed version for this specific CVE(s)
+      - How to fix that problem
+      - Linked CWE if available. 
+      - ...
+    placeholder: |
+      Examples (for GNU tar):
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: This is too much work for a single CVE and risks circumvention in the long run.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 17]
```diff
@@ -1,34 +1,94 @@
...
+        - CWE: CWE-125 Out-of-bound read
+      - (Other vulnerabilities) ...
   validations:
     required: true
+# The original `patches` is renamed to `Fixed version or patches`. If an package update is suggested the reporter should include the latest version and the earliest fixed version if applicable. 
 - type: textarea
   attributes:
-    label: 'PoC(s)'
-    description: 'Please list links to available PoCs (Proofs of Concept).'
+    label: 'Fixed versions or patches'
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: This was already done with title + label, this is not necessary.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml (hunk 1 above)
Comment (CamberLoid) [reply .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: (will remove when merging)

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml (hunk 17 above)
Comment (CamberLoid) [reply .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: May be called `solutions`? This field is intended to replace original `Patches`.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 18]
```diff
@@ -1,34 +1,81 @@
+# This is the draft of security vulnerabilities bug report, next version, revision 4.1
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml:1]: Drop lines 1-3 and move it to PR body. This is not necessary.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 19]
```diff
@@ -1,34 +1,81 @@
...
+# Severity is a new field of the template. This may help maintainers to identify which issue is most urgent and require attentions.
+# Note that a `None` indicates that the severity is under evaluation now, and not recommended.
+- type: dropdown
+  attributes:
+    label: 'Severity'
+    description: | 
+      Select the severity of reported vulnerability(ies). Refer to our wiki (not ready yet) for details.
+      Choose the highest severity out of reported vulnerabilities.
+    options:
+      - 'Critical/0-day'
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: These are not the same, list 0-day as the highest severity.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 20]
```diff
@@ -1,34 +1,81 @@
...
   validations:
     required: true
+# The original `patches` is renamed to `Fixed version or patches`. If an package update is suggested the reporter should include the latest version and the earliest fixed version if applicable. 
 - type: textarea
   attributes:
-    label: 'PoC(s)'
-    description: 'Please list links to available PoCs (Proofs of Concept).'
+    label: 'Patch(es)/Solution(s)'
+    description: | 
+      Enter N/A to indicate that a regular upgrade to latest version is favoured. Otherwise, fill this field with one or more of the following:
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Favored *. Pick one between American and British English.

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 21]
```diff
@@ -1,34 +1,81 @@
...
+      - Patch(es), if a backport is advised or no new upstream release is available
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: comma before or

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 22]
```diff
@@ -1,34 +1,81 @@
...
+      - Specific version, if upgrading to latest requires extra attention (please specify)
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: A specific non-latest version, if upgrading to the latest version may cause issue (please specify)

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 23]
```diff
@@ -1,34 +1,81 @@
...
+      - Workaround, if no fixes from upstream is released as the report submitted
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Other suggestions, such as a workaround/Errata documentation, if no fix is yet available

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml [hunk 24]
```diff
@@ -1,34 +1,81 @@
...
+    required: true
+# Severity is a new field of the template. This may help maintainers to identify which issue is most urgent and require attentions.
+- type: dropdown
   attributes:
-    label: 'Other security advisory IDs'
-    description: 'Other security advisory IDs. Please list with references to each ID.'
+    label: 'Severity'
+    description: | 
+      Select the severity of reported vulnerability(ies). Choose the highest severity out of reported vulnerabilities. For details about different severity level, you may refer to [SUSE's security flaws rating](https://www.suse.com/support/security/rating/). 
+      Additionally, `0-day` is used to indicate that the reporting vulnerabilities is being exploited for malicious activities. This is the highest severity rating regardless of its original severity.
```
Comment (MingcongBai) [inline .github/ISSUE_TEMPLATE/security-vulnerabilities-report.yml]: Emphasize, "is already being exploited"

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**